src/riak_ttb_codec.erl

```diff
@@ -58,21 +58,15 @@ return_resp({Atom, <<>>}) ->
 return_resp(Resp) ->
     Resp.
 
-de_stringify(Tuple) when is_tuple(Tuple) ->
-    list_to_tuple(de_stringify(tuple_to_list(Tuple)));
-de_stringify(List) when is_list(List), is_integer(hd(List)) ->
-    %% Yes, this could corrupt utf-8 data, but we should never, ever
-    %% have put it in string format to begin with
-    list_to_binary(List);
-de_stringify(List) when is_list(List) ->
-    lists:map(fun de_stringify/1, List);
-de_stringify(Element) ->
-    Element.
-
 encode_ts_rows(Rows) ->
     [encode_ts_row(Row) || Row <- Rows].
 
-encode_ts_row(Row) when is_list(Row) ->
-    list_to_tuple(Row);
 encode_ts_row(Row) when is_tuple(Row) ->
-    Row.
+    encode_ts_row(tuple_to_list(Row));
+encode_ts_row(Row) when is_list(Row) ->
+    list_to_tuple([undefined_to_empty(Cell) || Cell <- Row]).
+
+undefined_to_empty(undefined) ->
+    [];
+undefined_to_empty(Cell) ->
+    Cell.
```

Comment:
Before the TTB merge, the conversion from `undefined` to `[]` was done in `riak_pb_ts_codec:decode_cells` (https://github.com/basho/riak_pb/blob/riak_ts-develop/src/riak_pb_ts_codec.erl#L239), and the clients would see `[]` (for example, https://github.com/basho/riak_test/blob/riak_ts-develop/tests/ts_simple_describe_table.erl#L43).

After the TTB merge, functions from `riak_pb_ts_codec` are not used (there is virtually no codec at all), and `undefined`s pass through as is.

Given the fact that all NULL values originalting from `get_data` are already `[]`, the only source of `undefined`s, I suspect, is the code generating `[]` in DESCRIBE query code.

Let's don't do any conversions and fix the DESCRIBE code and see if that's all that needed to be done?